src/App.tsx

```diff
@@ -1,61 +1,136 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find(user => (user.id === userId)) || null
+);
+
+const getUserByName = (userName: string): User | null => (
+  usersFromServer.find(user => (user.name === userName)) || null
+);
+
+const getTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getTodos);
+  const [title, setTitle] = useState('');
+  const [userSelect, setUserSelect] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setIsTitleValid(true);
+    }
+
+    if (!userSelect) {
+      setIsUserValid(true);
+    }
+
+    if (!userSelect || !title) {
+      return;
+    }
+
+    setTodos((currentTodos) => {
+      const maxTodoId = Math.max(...currentTodos.map(todo => todo.id));
+      const newTodo = getUserByName(userSelect);
```

Comment:
when we change user we store his id on our state, can we remove `getUserByName` and start using `getUserById`?